test/multi-erc20.test.ts

```diff
@@ -0,0 +1,318 @@
+      .connect(owner)
+      .acceptBid(
+        signature,
+        params.bidNonce,
+        bidder.address,
+        params.bid,
+        params.tokenId,
+        params.minBid,
+        params.startBlock,
+        params.expireBlock
+      );
+    const receipt = await tx.wait();
+    expect(receipt.events?.length).to.eq(1); // BidAccepted
+    expect(receipt.from).to.eq(owner.address);
+  });
+  it("Fails when the token given matches the domain network token but not the token of the original bid", async () => {
+    const params = {
+      bidNonce: "4771690347",
+      bid: ethers.utils.parseEther("123"),
+      tokenId: "0x1",
+      minBid: "0",
+      startBlock: "0",
+      expireBlock: "999999999999",
+    };
+    fakeZNSHub.parentOf.whenCalledWith(dummyDomainId).returns("0");
+    fakeZNSHub.parentOf.whenCalledWith(params.tokenId).returns(dummyDomainId);
+    fakeZNSHub.ownerOf.whenCalledWith(params.tokenId).returns(owner.address);
+    fakeRegistrar.minterOf.whenCalledWith("0x1").returns(owner.address);
+
+    const bidToSign = await zAuction.createBid(
+      params.bidNonce,
+      params.bid,
+      params.tokenId,
+      params.minBid,
+      params.startBlock,
+      params.expireBlock,
+      fakeNetworkToken.address
+    );
+
+    const signature = await bidder.signMessage(
+      ethers.utils.arrayify(bidToSign)
+    );
+
+    const tx = zAuction
+      .connect(owner)
+      .acceptBidV2(
+        signature,
+        params.bidNonce,
+        bidder.address,
+        params.bid,
+        params.tokenId,
+        params.minBid,
+        params.startBlock,
+        params.expireBlock,
+        fakeDefaultToken.address
+      );
+
+    await expect(tx).to.be.revertedWith("zAuction: recovered incorrect bidder");
+  });
+  it("Accepts a bid when the right token is used", async () => {
+    const params = {
+      bidNonce: "5771690347",
+      bid: ethers.utils.parseEther("123"),
+      tokenId: "0x1",
+      minBid: "0",
+      startBlock: "0",
+      expireBlock: "999999999999",
+    };
+    fakeZNSHub.parentOf.whenCalledWith(params.tokenId).returns(dummyDomainId);
+    fakeZNSHub.ownerOf.whenCalledWith(params.tokenId).returns(owner.address);
+    const bidToSign = await zAuction.createBid(
+      params.bidNonce,
+      params.bid,
+      params.tokenId,
+      params.minBid,
+      params.startBlock,
+      params.expireBlock,
+      fakeDefaultToken.address
+    );
+    const signature = await bidder.signMessage(
+      ethers.utils.arrayify(bidToSign)
+    );
+    const tx = await zAuction
+      .connect(owner)
+      .acceptBidV2(
+        signature,
+        params.bidNonce,
+        bidder.address,
+        params.bid,
+        params.tokenId,
+        params.minBid,
+        params.startBlock,
+        params.expireBlock,
+        fakeDefaultToken.address
+      );
+    const receipt = await tx.wait();
+    expect(receipt.events?.length).to.eq(1); // BidAccepted
+    expect(receipt.from).to.eq(owner.address);
```

Comment:
This really doesn't check anything worthwhile since the tx receipt is always going to be from the person who sent the tx.